diesel/Cargo.toml

```diff
@@ -17,6 +17,7 @@ byteorder = "0.3.*"
 quickcheck = { version = "0.2.25", optional = true }
 chrono = { version = "^0.2.17", optional = true }
 uuid = { version = "^0.2.0", optional = true, features = ["use_std"] }
+log = { version = "0.3", optional = true }
```

Comment:
You should be able to filter the logging statement on crate basis. For example if someone uses [env_logger](https://github.com/rust-lang/log) as logging backend one could set RUST_LOG=diesel=debug to output all debugging statements from diesel with a log level of debug or higher. Also you could implement your own backend to filter on custom rules.
If you don't like this approach I will change this as suggested. I've choosen the `log` crate because it seems to be the standard for such things and seems to be greatly configurable form the users of this library.